backend/src/modules/user/repositories/user-repository.service.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { UserEntity } from '../entities/user.entity';
+import { IUserRepository } from '../interfaces/user-repository.interface';
+import { IUserModel } from '../models/user.model';
+
+@Injectable()
+export class UserRepositoryService implements IUserRepository {
+  constructor(
+    @InjectRepository(UserEntity)
+    private readonly userRepository: Repository<UserEntity>,
+  ) {}
+
+  create(userModel: Omit<IUserModel, 'id'>): Promise<IUserModel> {
```

Comment:
`Create` cred ca ar trebui sa faca doar `AdminUser` sau `VolunteerUser` (facem overload la metoda), `UserEntity` nu are sens singur. (D'aia ma gandeam daca mai are sens sa-l punem in DB sau il facem abstract si pastram cele 2 entitati)